--- origsrc/imsettings-1.6.8/data/imsettings-functions.in	2015-07-23 02:23:02.225937100 -0500
+++ src/imsettings-1.6.8/data/imsettings-functions.in	2015-07-23 04:56:20.476466600 -0500
@@ -119,7 +119,7 @@ function get_desktop() {
 
 function is_gtk_supported() {
     [ -n "$IMSETTINGS_DISABLE_DESKTOP_CHECK" ] && return 0
-    case "$(get_desktop|tr '[A-Z]' '[a-z]')" in
+    case "$(get_desktop|tr '[:upper:]' '[:lower:]')" in
 	cinnamon*)
 	    if check_imsettings_capability cinnamon-gsettings; then
 		return 0
@@ -154,7 +154,7 @@ function is_gtk_supported() {
 
 function is_qt_supported() {
     [ -n "$IMSETTINGS_DISABLE_DESKTOP_CHECK" ] && return 0
-    case "$(get_desktop|tr '[A-Z]' '[a-z]')" in
+    case "$(get_desktop|tr '[:upper:]' '[:lower:]')" in
 	*)
 	    if check_imsettings_capability qt; then
 		return 0
--- origsrc/imsettings-1.6.8/data/imsettings-target-checker.sh.in	2015-11-20 11:53:10.088411500 -0600
+++ src/imsettings-1.6.8/data/imsettings-target-checker.sh.in	2015-11-20 11:54:07.982263100 -0600
@@ -75,7 +75,7 @@ gsettings_get_bool() {
     fi
 }
 
-case $(get_desktop|tr '[A-Z]' '[a-z]') in
+case $(get_desktop|tr '[:upper:]' '[:lower:]') in
     *gnome|gnome*)
 	dbus_send 'org.gnome.Shell' '/' 'org.freedesktop.DBus.Peer.Ping'
 	if [ $? = 0 ]; then
